infra/templates/az-isolated-service-single-region/app.tf

```diff
@@ -33,3 +33,76 @@ module "keyvault" {
     "azurerm" = "azurerm.app_dev"
   }
 }
+
+module "container_registry" {
+  source                  = "../../modules/providers/azure/container-registry"
+  container_registry_name = local.acr_name
+  resource_group_name     = azurerm_resource_group.app_rg.name
+  // Note: this is requird until App Services and ACR work over MSI. See the design document for more details.
+  container_registry_admin_enabled = true
+  // Note: only premium ACRs allow configuration of network access restrictions
+  container_registry_sku = "Premium"
+  providers = {
+    "azurerm" = "azurerm.app_dev"
+  }
+}
+
+# Build and push the app service image slot to enable continuous deployment scenarios.
+# We're using ACR build tasks to remotely carry out the docker build / push.
+resource "null_resource" "acr_image_deploy" {
+  count      = length(var.deployment_targets)
+  depends_on = [module.container_registry]
+
+  triggers = var.deployment_targets[count.index]
+
+  provisioner "local-exec" {
+    command = "az acr build -t $IMAGE_TAG -r $DOCKER_REPOSITORY -f $DOCKER_FILE $GIT_SOURCE"
```

Comment:
Have you explored what's involved with leveraging acr build tasks to source the personal access token for private hld repos?

Reply:
ACR Build tasks do not seem to work with private repositories [Github Issue](https://github.com/MicrosoftDocs/azure-docs/issues/7975)

However, the source location can be local. Assuming we are able to pull the source down *somehow* we should be able to specify this with the last argument:

```
<SOURCE_LOCATION>
The local source code directory path (e.g., './src') or the URL to a git repository (e.g., 'https://github.com/Azure-Samples/acr-build-helloworld-node.git') or a remote tarball (e.g., 'http://server/context.tar.gz').
```

Noob question: I thought a PAT was only needed to push *into* a repo. Whether or not we can pull into the repo is based on the repos visibility (public|private). Is this correct?